ts/components/LabelledItem.tsx

```diff
@@ -28,11 +35,12 @@ const styles = StyleSheet.create({
   }
 });
 
-type StyleType = IconProps["style"];
+// Style type must be generalized so as to support both text icons and image icons
+type StyleType = IconProps["style"] & ImageStyle;
 
 type CommonProp = Readonly<{
   label: string;
-  icon?: string;
+  icon?: string & ImageSourcePropType;
```

Comment:
```suggestion
  icon?: string | ImageSourcePropType;
```
It should be a `string` or an `ImageSourcePropType`.
Is it right?

Reply:
In the limited case of AddCardScreen you are probably right. But union types defined with the '|' operator doesn't allow access to properties and methods the composed types have in common. Intersection types, defined with the '&' operator, instead allows access to all methods and properties of the composed types. It's a bit counterintuitive. 
As I didn't know in advance whether the composed types had a common ancestor or not, I chose the most general option. Obviously I could change to have a stricter type checking

Reply:
@GiovanniMancini 
Ok, the goal is clear but I still don't get a scenario where you define a value that can be a `string` and a `ImageSourcePropType` at the same time 

```
export type ImageSourcePropType = ImageURISource | ImageURISource[] | ImageRequireSource;

export interface ImageURISource {
    /**
     * `uri` is a string representing the resource identifier for the image, which
     * could be an http address, a local file path, or the name of a static image
     * resource (which should be wrapped in the `require('./path/to/image.png')`
     * function).
     */
    uri?: string;
    /**
     * `bundle` is the iOS asset bundle which the image is included in. This
     * will default to [NSBundle mainBundle] if not set.
     * @platform ios
     */
    bundle?: string;
    /**
     * `method` is the HTTP Method to use. Defaults to GET if not specified.
     */
    method?: string;
    /**
     * `headers` is an object representing the HTTP headers to send along with the
     * request for a remote image.
     */
    headers?: { [key: string]: string };
    /**
     * `cache` determines how the requests handles potentially cached
     * responses.
     *
     * - `default`: Use the native platforms default strategy. `useProtocolCachePolicy` on iOS.
     *
     * - `reload`: The data for the URL will be loaded from the originating source.
     * No existing cache data should be used to satisfy a URL load request.
     *
     * - `force-cache`: The existing cached data will be used to satisfy the request,
     * regardless of its age or expiration date. If there is no existing data in the cache
     * corresponding the request, the data is loaded from the originating source.
     *
     * - `only-if-cached`: The existing cache data will be used to satisfy a request, regardless of
     * its age or expiration date. If there is no existing data in the cache corresponding
     * to a URL load request, no attempt is made to load the data from the originating source,
     * and the load is considered to have failed.
     *
     * @platform ios
     */
    cache?: 'default' | 'reload' | 'force-cache' | 'only-if-cached';
    /**
     * `body` is the HTTP body to send with the request. This must be a valid
     * UTF-8 string, and will be sent exactly as specified, with no
     * additional encoding (e.g. URL-escaping or base64) applied.
     */
    body?: string;
    /**
     * `width` and `height` can be specified if known at build time, in which case
     * these will be used to set the default `<Image/>` component dimensions.
     */
    width?: number;
    height?: number;
    /**
     * `scale` is used to indicate the scale factor of the image. Defaults to 1.0 if
     * unspecified, meaning that one image pixel equates to one display point / DIP.
     */
    scale?: number;
}

export type ImageRequireSource = number;
```

afaik, if you have

```typescript
type Person = {name: string};
type Pet = {nickname: string};

type PPInterection = Person & Pet;
const test = {name: "Robert", nickname: "Bob"}
```
you can take advantage of both types.

But in a scenario like (similar to `string & ImageSourcePropType`)

```typescript
type NumberAndString = number & string;
```
I can define a value that is a string or number, there is no overlap. (both have `toString` method for example, but I guess this is not the point)

> It's a bit counterintuitive.
`icon` can be a _string_ (vector case) OR an _image source_ (asset case). Isn't it?

Maybe I'm losing something but I can see [an advantage](https://it.wikipedia.org/wiki/You_aren%27t_gonna_need_it).
Could you show an example when the type `string & ImageSourcePropType` is useful?

Reply:
In my opinion number & string is different from string & ImageSourceType. In the first case the composed types are basic types, so they are guaranteed to be disjoint, in the second case there is not an a-priori guarantee. Despite my previous idea, the type definitions you posted makes clear that composed types are disjoint. Moreover the react components to which the IconSource is passed down make an implicit cast to the right type, so the IconSource is never used as is. My solution is too general and unneeded, so I agree.. let's switch to union type. Thank you for pointing out such a subtle behavior!